packages/components/src/box-control/all-input-control.tsx

```diff
@@ -70,17 +94,34 @@ export default function AllInputControl( {
 	};
 
 	return (
-		<UnitControl
-			{ ...props }
-			disableUnits={ isMixed }
-			isOnly
-			value={ allValue }
-			onChange={ handleOnChange }
-			onUnitChange={ handleOnUnitChange }
-			onFocus={ handleOnFocus }
-			onHoverOn={ handleOnHoverOn }
-			onHoverOff={ handleOnHoverOff }
-			placeholder={ allPlaceholder }
-		/>
+		<HStack>
+			<UnitControl
+				{ ...props }
+				disableUnits={ isMixed }
+				id={ inputId }
+				isOnly
+				value={ allValue }
+				onChange={ unitControlOnChange }
+				onUnitChange={ handleOnUnitChange }
+				onFocus={ handleOnFocus }
+				onHoverOn={ handleOnHoverOn }
+				onHoverOff={ handleOnHoverOff }
+				placeholder={ allPlaceholder }
+			/>
+
+			<FlexedRangeControl
+				__nextHasNoMarginBottom
+				aria-controls={ inputId }
+				aria-labelledby={ inputId }
```

Comment:
Sorry, we still need to give this RangeControl the same `label` as the UnitControl. Just the way we were doing it with `aria-labelledby` was invalid.

Reply:
Ah, makes sense. Thanks for clarifying!

Reply:
It was awhile ago, but I just recalled a previous [related discussion](https://github.com/WordPress/gutenberg/pull/56665/commits/94db8094068828ac1e22a3cb3ed71d949c263bd2#r1420981268). We initially had another label on `RangeControl` but we weren't sure about it since the slider is a redundant control. 

So that's why I tried to implement `aria-labelledby={ inputId }` instead, but as you pointed out, it wasn't valid since it pointed to the `inputId` rather than to a label. 

But if we add the label to `RangeControl`, we end up with two of the same labels in the DOM.

So I'm thinking it might make sense to add a label with `<VisuallyHidden>`, and then we can use `aria-labelledby` to indicate both controls are for the same input with the one label.  Like:

```diff
diff --git a/packages/components/src/box-control/all-input-control.tsx b/packages/components/src/box-control/all-input-control.tsx
index 8dcefc118b..30ec2c0396 100644
--- a/packages/components/src/box-control/all-input-control.tsx
+++ b/packages/components/src/box-control/all-input-control.tsx
@@ -18,6 +18,7 @@ import {
 	isValuesDefined,
 	CUSTOM_VALUE_SETTINGS,
 } from './utils';
+import { VisuallyHidden } from '../visually-hidden';
 
 const noop = () => {};
 
@@ -73,6 +74,13 @@ export default function AllInputControl( {
 
 	return (
 		<HStack>
+			<VisuallyHidden
+				as="label"
+				htmlFor={ inputId }
+				id={ `${ inputId }-label` }
+			>
+				{ LABELS.all }
+			</VisuallyHidden>
 			<UnitControl
 				{ ...props }
 				className="component-box-control__unit-control"
@@ -84,13 +92,12 @@ export default function AllInputControl( {
 				onUnitChange={ handleOnUnitChange }
 				onFocus={ handleOnFocus }
 				placeholder={ allPlaceholder }
-				label={ LABELS.all }
-				hideLabelFromVision
 			/>
 
 			<FlexedRangeControl
 				__nextHasNoMarginBottom
-				aria-controls={ inputId }
+				aria-labelledby={ `${ inputId }-label` }
 				onChange={ sliderOnChange }
 				min={ 0 }
 				max={ CUSTOM_VALUE_SETTINGS[ parsedUnit ?? 'px' ]?.max ?? 10 }

```

Thoughts?

Reply:
I don't think there is any benefit in the two inputs being associated to the same `label` node, aside from the number of characters it may or may not save in the bundle. I wouldn't mind it if it weren't adding complexity to the codebase, but in this case I think it does add unnecessary complexity by not using the labeling mechanisms that are already built into UnitControl and RangeControl.

I would do it the simple way (`aria-label`, or `label` + `hideLabelFromVision`), similar to how we do it in RangeControl or FontSizePicker.

Reply:
> I don't think there is any benefit in the two inputs being associated to the same label node... I would do it the simple way.

Agreed that we should proceed in whatever way makes the most sense from a development perspective here. The important thing is that the fields are labelled correctly, so as long as that's the case, if it takes two labels, it takes two labels.